src/Compilers/CSharp/Test/Semantic/Semantics/BindingAsyncTests.cs

```diff
@@ -3529,6 +3529,57 @@ async Task<int> F1()
                 Diagnostic(ErrorCode.ERR_SynchronizedAsyncMethod, "F1"));
         }
 
+        [Fact, WorkItem(66530, "https://github.com/dotnet/roslyn/issues/66530")]
+        public void BadAsync_MethodImpl_Synchronized_Lambda()
+        {
+            var source = @"
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+public class C
+{
+    void M()
+    {
+        _ = [MethodImpl(MethodImplOptions.Synchronized)] async Task<int> () =>
```

Comment:
Shouldn't we test `var lambda = [MethodImpl(MethodImplOptions.Synchronized)] async Task<int> () =>` instead to avoid error "CS8183: Cannot infer the type of implicitly-typed discard." and instead demonstrate that error "CS4015: 'MethodImplOptions.Synchronized' cannot be applied to an async method" will be reported?

Reply:
D'oh. Indeed. Thanks